src/bsp/reboot.c

```diff
@@ -0,0 +1,73 @@
+#include "reboot.h"
+
+#include <pthread.h>
+#include <signal.h>
+#include <string.h>
+#include <time.h>
+
+#include <xss.h>
+#include <zephyr/drivers/pm_cpu_ops.h>
+
+static timer_t timer_id;
+
+static void reboot(sigval arg)
+{
+    int rc;
+    static const char* path = CONFIG_AOS_REBOOT_XEN_STORE_PATH;
+    char value[CONFIG_AOS_REBOOT_XEN_STORE_VALUE_LEN] = { 0 };
+
+    printk("Check reboot state");
+
+    rc = xss_read(path, value, sizeof(value)); 
+    if (rc)
+    {
+        printk("failed to read reboot value (%d)\n", rc);
+
+        return;
+    }
+
+    if (strcmp(value, "2"))
+    {
+        printk("unexpected reboot value (%s)\n", value);
+
+        return;
+    }
+
+    printk("Rebooting ....");
+
+    rc = pm_system_reset(SYS_COLD_RESET);
+    if (rc)
+    {
+        printk("failed to reboot system (%d)\n", rc);
+    }
+}
+
+int start_reboot_monitoring()
+{
+    int ret;
+    int intervalMs = CONFIG_AOS_REBOOT_CHECKING_TIMEOUT;
```

Comment:
Add define in header file instead of config.

Reply:
Done